mm2src/coins/eth.rs

```diff
@@ -6218,6 +6248,21 @@ pub async fn eth_coin_from_conf_and_request(
     if swap_contract_address == Address::default() {
         return ERR!("swap_contract_address can't be zero address");
     }
+    let maker_swap_v2_contract: Address = try_s!(json::from_value(req["maker_swap_v2_contract"].clone()));
+    if maker_swap_v2_contract == Address::default() {
+        return ERR!("maker_swap_v2_contract can't be zero address");
+    }
+    let taker_swap_v2_contract: Address = try_s!(json::from_value(req["taker_swap_v2_contract"].clone()));
+    if taker_swap_v2_contract == Address::default() {
+        return ERR!("taker_swap_v2_contract can't be zero address");
+    }
+    let nft_maker_swap_v2_contract: Option<Address> =
+        try_s!(json::from_value(req["nft_maker_swap_v2_contract"].clone()));
+    if let Some(nft_maker_swap_v2) = nft_maker_swap_v2_contract {
+        if nft_maker_swap_v2 == Address::default() {
+            return ERR!("nft_maker_swap_v2_contract can't be zero address");
+        }
+    }
```

Comment:
I prefer not to do swap_2 contracts optional since this will be the main swap protocol, when we finish it. 
I made nft swap contract optional, as nft is kinda optional feature for users.
Lets wait for @shamardy decision

Reply:
@mariocynicys @shamardy 
did activation changes in this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129/commits/1e906003a3ab70845ef91a6f33bab12bbaf59998

Added `swap_v2_contracts: Option<SwapV2Contracts>` field in EthCoin.

In the changes, the legacy "enable" RPC ignores swaps v2 and sets "swap_v2_contracts" as none.

For v2 activation rpc, it checks if "use_trading_proto_v2" is true. if so, "swap_v2_contracts" must exist.

Reply:
req example
```json
curl --url "http://127.0.0.1:7783" --data '{
  "userpass": "'$USERPASS'",
  "method": "enable_eth_with_tokens",
  "mmrpc": "2.0",
  "params": {
    "ticker": "MATIC",
    "mm2": 1,
    "swap_contract_address": "0x9130b257D37A52E52F21054c4DA3450c72f595CE",
    "swap_v2_contracts":{
	    "maker_swap_v2_contract": "0x9130b257D37A52E52F21054c4DA3450c72f595CE",
	    "taker_swap_v2_contract": "0x9130b257D37A52E52F21054c4DA3450c72f595CE",
	    "nft_maker_swap_v2_contract": "0x9130b257D37A52E52F21054c4DA3450c72f595CE"
    },    
    "fallback_swap_contract": "0x9130b257D37A52E52F21054c4DA3450c72f595CE",
    "nodes": [
      {
        "url": "https://polygon-rpc.com"
      }
    ],
    "erc20_tokens_requests": [],
    "nft_req": {
      "provider":{
        "type": "Moralis",
        "info": {
          "url":"http://localhost:6150/nft-test",
          "proxy_auth":true
        }
      }
    }
  }
}'
```

Reply:
Nice. But don't we be better off having `nft_maker_swap_v2_contract` wrapped in another optional for simplicity.

Reply:
> Nice. But don't we be better off having `nft_maker_swap_v2_contract` wrapped in another optional for simplicity.

By simplicity do you mean request json format or code maintainability?

If we move `nft v2 contracts` to a separate structure, it means we will need to perform additional check in v2 activation. Currently, we check `conf["use_trading_proto_v2"]` here: 
https://github.com/KomodoPlatform/komodo-defi-framework/blob/4e9f90248f15989bf0bb3ca8f270572465057635/mm2src/coins/eth/v2_activation.rs#L600 

Moving `nft_maker_swap_v2_contract` to a separate structure requires adding the `conf["use_nft_trading_proto_v2"]` flag in the MM config. If user decides to do the NFT swap, they have to restart the application with the updated flag.

Additionally, this complicates the start swap process in `lp_ordermatch.rs`. Let's take Bob (Maker) functionality as an example.
Here, we check if MM conf has `TPUV2` (trading proto up v2) set to true.
https://github.com/KomodoPlatform/komodo-defi-framework/blob/4e9f90248f15989bf0bb3ca8f270572465057635/mm2src/mm2_main/src/lp_ordermatch.rs#L2973-L2982
If we add `if ctx.use_nft_trading_proto_v2()`, it means that we need to handle `MakerSwapStateMachine` for NFTs differently / separately? However, our goal is to have a general swap state machine implementation for both coins and NFTs.

I doubt that this approach will simplify things. If the swap state machine code should be common for both coin and NFT features, it's better to keep V2 smart contract addresses in a common structure.

But if you have a different view, please share it.